features/run/service_endpoint_environment_variables.feature

```diff
@@ -0,0 +1,29 @@
+Feature: Service endpoint enviornment variable
+
+  As an Exosphere developer
+  I want the services to know where each other are available from the outside world
+  So clients can talk directly to other services
+
+  Rules:
+  - "SERVICE_<NAME>_EXTERNAL_ORIGIN" is passed as an environment variable to
+    each service for any public service
+
+
+  Scenario: external origin env var available at run time
+    Given I am in the root directory of the "service-exteral-origin-env" example application
+    And starting "exo run" in my application directory
+    And it prints "Listening on port 3000" in the terminal
+    And it prints "webpack: Compiled successfully" in the terminal
+    Then http://localhost:3010 displays:
+      """
+      Backend located at http://localhost:3000
+      """
```

Comment:
I'm starting to think that having to manually write up an example app per feature file may start to get unmaintainable quickly.

Do you think it would be possible to have feature files that self-describe their appllications on a higher level and we can have step definitions scaffold them out in a temp directory?

Maybe something like
```gherkin
Scenario: external origin env var available at run time
  Given I have an application with the following services:
    | NAME     | TYPE |
    | frontend | web  |
    | api      | web  |
  When running the web service
  Then I have the following environment variables available at run time:
    | NAME                             | VALUE                 |
    | SERVICE_FRONTEND_EXTERNAL_ORIGIN | http://localhost:3000 |
    | SERVICE_API_EXTERNAL_ORIGIN      | http://localhost:3010 |
```

Reply:
I do like that the example apps are there for real use-cases and I don't think we should remove them fully, but we should have a small manageable amount of them showing large scale features (like communicating with exocom perhaps). But there are a lot of smaller more edge-casey features that I think should be more like this style.

As for implementation, our scaffolded services could prepare all the possible information that we might care about and expose them through a web interface or something. For example, we could have a template example app that get's scaffolded for each service under `Given I have an application with the following services:`.

That service can expose whatever data we need through http, for example, `http://host/build-time-env-vars` could return the environment variables that were availalbe at build time.
This can be done with something like
```dockerfile
RUN env > /usr/app/public/build-time-env-vars
```

and we could do something similar for run-time env vars `http://host/run-time-env-vars`

What are your thoughts?